### ISSUE TITLE
[java] A false negative about the rule MissingOverride

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#3541]
**Affects PMD Version:6.39.0** 

**Rule:MissingOverride**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/pmd-6.39.0/pmd_rules_java_bestpractices.html#missingoverride>

**Description:**
Based on the following test case, PMD should have reported a warning on line 2 because this class has a method **toString** which needs to be annotated with **Override**. I think this case is easy and should be reported.

**Code Sample demonstrating the issue:**

```
public class Worker {
    public String toString() { // should report a warning here
        return "123456";  
    }
}
```

**Expected outcome: A Warning**

PMD should report a violation at line 2, but no warnings. This is a false negative.

**Running PMD through:** *[Maven]*
